pyop2/codegen/builder.py

```diff
@@ -442,6 +453,91 @@ def emit_unpack_instruction(self, *, loop_indices=None):
         yield call
 
 
+class MixedMatPack(Pack):
+
+    insertion_names = {False: "MatSetValuesBlockedLocal",
+                       True: "MatSetValuesLocal"}
+    """Function call name for inserting into the PETSc Mat. The keys
+       are whether or not maps are "unrolled" (addressing dofs) or
```

Comment:
What keys? The insertion_names?

Reply:
`False` and `True`. In fact, this is unnecessary since a `MixedMatPack` never inserts directly (instead inserting into the pieces).